datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -211,9 +287,29 @@ pub(crate) fn min_statistics<'a, I: Iterator<Item = Option<&'a ParquetStatistics
     data_type: &DataType,
     iterator: I,
 ) -> Result<ArrayRef> {
-    let scalars = iterator
-        .map(|x| x.and_then(|s| get_statistic!(s, min, min_bytes, Some(data_type))));
-    collect_scalars(data_type, scalars)
+    match data_type {
```

Comment:
Here is an example of how they work -- you make `BoolStatsIterator::new(iterator)` which then returns a sequence of `Option<&bool>` and then the min/max can be extracted out